petab/core.py

```diff
@@ -770,3 +770,66 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
```

Comment:
```suggestion
        petab_problem: Problem) -> Problem:
```

Please also adapt the documentation for the return value, now that we are returning something.

That the current implementation of the test passes, means that we are still modifying the original object. If we keep doing that, we should not return anything. Otherwise, we should only modify a copy. Either way should be stated in the docstring.

Reply:
I return the object `Problem` as changes in the sbml model and in the measurement dataframe occur.
Could you please clarify ` If we keep doing that, we should not return anything` ?

Reply:
I included now a short explanation

Reply:
> I return the object Problem as changes in the sbml model and in the measurement dataframe occur.

Exactly, you are modifying the model and the dataframe, but not the Problem instance (the references stay the same, only the pointed-to objects are changed).

So option A: 
- Modify the original dataframe and model, and don't return anything. This makes it the user's responsibility to keep a copy if required. Returning Problem is not necessary, because is not changed as mentioned above.

Option B:
- Create a deep copy of problem, do all modification on that, and return the copy.

Reply:
Actually options A sounds good! Thanks for the suggestion.
So I should just replace the original file with the new one?

Reply:
For option A, everything was fine, except for the return statement that should be removed.